TrafficCapture/testingEnvironment/build.gradle

```diff
@@ -0,0 +1,125 @@
+plugins {
+    id 'org.opensearch.migrations.java-library-conventions'
+    id "com.avast.gradle.docker-compose" version "0.16.12"
+    id "com.bmuschko.docker-java-application" version "9.3.1"
+}
+
+import java.security.MessageDigest
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+import org.apache.tools.ant.taskdefs.condition.Os
+
+def calculateDockerHash(String projectName) {
+    MessageDigest digest = MessageDigest.getInstance('SHA-256')
+    fileTree("src/main/docker/${projectName}")
+            .each( file ->
+                    file.withInputStream { is ->
+                        var buffer = new byte[1024]
+                        int read
+                        while ((read = is.read(buffer)) != -1) {
+                            digest.update(buffer, 0, read)
+                        }
+                    }
+            )
+    return digest.digest().encodeHex().toString()
+}
+
+dependencies {
+    implementation project(':trafficCaptureProxyServer')
+    testImplementation project(':trafficCaptureProxyServerTest')
+}
+
+def dockerFilesForExternalServices = [
+        "nginx": "nginx"
+]
+
+// Create the static docker files that aren't hosting migrations java code from this repo
+dockerFilesForExternalServices.each { projectName, dockerImageName ->
+    task("buildDockerImage_${projectName}", type: DockerBuildImage) {
+        def hash = calculateDockerHash(projectName)
+        images.add("testenv/${dockerImageName}:$hash")
+        images.add("testenv/${dockerImageName}:latest")
+        inputDir = project.file("src/main/docker/${projectName}")
+    }
```

Comment:
This should be refactored into a common shared class and imported by both projects